src/main/scala/com/gu/newsletterlistcleanse/braze/BrazeClient.scala

```diff
@@ -0,0 +1,52 @@
+package com.gu.newsletterlistcleanse.braze
+
+import org.slf4j.{Logger, LoggerFactory}
+import scalaj.http._
+import io.circe.parser.decode
+import io.circe.syntax._
+
+class BrazeClient {
+
+  private val timeout = 5000
+
+  val logger: Logger = LoggerFactory.getLogger(this.getClass)
+  val brazeEndpoint = "https://rest.fra-01.braze.eu"
+
+  private def withClientLogging[A](info: => String)(block: => Either[BrazeError,A]): Either[BrazeError, A] = {
+    val result = block
+
+    result.foreach { successResult =>
+      logger.info(s"BrazeClient success $info $successResult")
+    }
+
+    result.left.foreach { errorResult =>
+      logger.error(s"BrazeClient failure $info $errorResult")
+    }
+
+    result
+  }
+
+  private def parseValidateResponse(response: HttpResponse[String]): Either[BrazeError, BrazeResponse] = {
+    decode[BrazeResponse](response.body) match {
+      case Right(parsedResponse) if response.is2xx  =>
+        Right(parsedResponse)
+      case Left(e) if response.is2xx =>
+        logger.error("failure to parse Braze response", e)
+        Left(BrazeError(response))
+      case _ =>
+        Left(BrazeError(response))
+    }
+  }
+
+  def updateUser(request: UserTrackRequest): Either[BrazeError, BrazeResponse] = {
+    withClientLogging(s"updating user: ${request.attributes.toString()}, ${request.events.toString()}"){
```

Comment:
we may want to make sure we don't dump too much personal data in the logs here

Reply:
This should be done now. I've removed specifics from the code and we just need to think about what we want to log once it's at PROD e.g., do we want to log which user broke? And how?